sdk/eventhub/azure-eventhub/tests/conftest.py

```diff
@@ -110,16 +110,38 @@ def timeout_factor(uamqp_transport):
 def fake_span():
     return FakeSpan
 
+
+@pytest.fixture(scope="session")
+def get_credential():
+    use_pwsh = os.environ.get("AZURE_TEST_USE_PWSH_AUTH", "false")
+    use_cli = os.environ.get("AZURE_TEST_USE_CLI_AUTH", "false")
+
+    credential = EnvironmentCredential()
+    # User-based authentication through Azure PowerShell, if requested
+    if use_pwsh.lower() == "true":
+        log.info(
+            "Environment variable AZURE_TEST_USE_PWSH_AUTH set to 'true'. Using AzurePowerShellCredential."
+        )
+        from azure.identity import AzurePowerShellCredential
+        credential = AzurePowerShellCredential()
+    # User-based authentication through Azure CLI, if requested
+    if use_cli.lower() == "true":
```

Comment:
If we continue to `return` the credential after these blocks, I would change this to `elif` so that we don't run through both the PWSH and CLI conditions